### DISCUSSION TITLE
How to Add Multiple Tilesets in a Tiledefs File

### DISCUSSION THREAD

**Discussion by BuzzLEquerre** [flareteam/flare-game#964] [General]
Hello, I'm desperately trying to add dungeon tiles in my mine map, only there is an 8 tile offset and I don't know why..
- I tried to integrate the default tilesetdef of Empyrain Campaing "tileset_cave_and_dungeon" only all the bug tiles, once in game it's a black screen.
- So I tried to add the tilesetdef "tileset_dungeon" to my tileset mine, and this time I have a shift in the game of the "610" tile of 8 tiles. I don't know what to do, I reread all my code, there is no lag.

Anyone know what to do? or how to do it, because I'm totally stuck there :/
Here's the lag at play, and what there should be :
![Capture d'écran 2024-08-11 162449](https://github.com/user-attachments/assets/6ba140c5-aaa3-4e65-94da-08ca436bcdd5)
![Capture d'écran 2024-08-11 162542](https://github.com/user-attachments/assets/4bae8b78-33da-4344-8d05-a039c0e44fae)
[tileset_cave.txt](https://github.com/user-attachments/files/16574028/tileset_cave.txt)

**Comment by dorkster** [comment]
If you look at the dungeon tileset in Tiled, you can see that there are some blank spaces. This causes the tile IDs to skip ahead in some places. For example, at ID 595, the next tile should be 600. It helps to have a text editor that can increment/decrement by a single value over several rows (VIM can do this, but there's probably others).

Try this. The tile IDs should be fixed:
[tileset_cave.txt](https://github.com/user-attachments/files/16575519/tileset_cave.txt)

**Reply by BuzzLEquerre** [reply]
![Capture d'écran 2024-08-11 192725](https://github.com/user-attachments/assets/990213b0-c324-4263-8213-4ffb1ac0073d)

what!! it's incredible, it works! How it was enough to just shift the tiles!!
Thank you so much for correcting my tilesetdef, it must have taken you so long! A huge thank you! I had spent more than 5 hours xd
